usecases/evaluate_scenario/evaluate_scenario.go

```diff
@@ -144,16 +180,10 @@ func EvalScenario(
 		Score:               score,
 		Outcome:             outcome,
 		OrganizationId:      params.Scenario.OrganizationId,
+		PivotValue:          pivotValue,
 	}
 	if params.Pivot != nil {
-		pivotValue, err := getPivotValue(ctx, *params.Pivot, dataAccessor)
-		if err != nil {
-			return models.ScenarioExecution{}, errors.Wrap(
-				err,
-				"error getting pivot value in EvalScenario")
-		}
 		se.PivotId = &params.Pivot.Id
```

Comment:
I would keep all pivot related stuff closed to each other (for readability)
```suggestion
		se.PivotId = &params.Pivot.Id
		se.PivotValue = pivotValue
```

Reply:
so instead of line 183 you mean ? Sure, can do that